File: orchestrator/moontide/pipeline.py
import logging

import requests

from moontide import config

_logger = logging.getLogger(__name__)

SAMPLE_CONTACT = {
    "name": "Alex Johnson",
    "company": "Acme Corp",
    "email": "alex.johnson@acme.com",
}


def run_once() -> None:
    """Run one pass of the enrichment pipeline."""
    contact = SAMPLE_CONTACT.copy()

    # Step 1: Validate
    try:
        resp = requests.post(config.FLUX_URL, json=contact, timeout=10)
        resp.raise_for_status()
        contact = resp.json()
    except Exception as exc:
        _logger.warning("validate failed: %s", exc)

    # Step 2: Enrich
    try:
        resp = requests.post(config.RIFT_URL, json=contact, timeout=10)
        resp.raise_for_status()
        contact = resp.json()
    except Exception as exc:
        _logger.warning("enrich failed: %s", exc)

    # Step 3: Score
    try:
        resp = requests.post(config.SWELL_URL, json=contact, timeout=10)
        resp.raise_for_status()
    except Exception as exc:
        _logger.warning("score failed: %s", exc)


def run(stop_event) -> None:
    """Run the pipeline loop until stop_event is set."""
    while not stop_event.is_set():
        run_once()
        stop_event.wait(config.PIPELINE_INTERVAL_SECONDS)

File: orchestrator/Pipeline.cs
using System.Net.Http.Json;

namespace MoonTide;

public record Contact(
    string Name = "",
    string Company = "",
    string Email = "",
    string NormalizedEmail = "",
    bool Valid = false,
    string Industry = "",
    string Size = "",
    string Region = "",
    int Score = 0,
    string[]? Tags = null
);

public class Pipeline
{
    private static readonly Contact SampleContact = new(
        Name: "Alex Johnson",
        Company: "Acme Corp",
        Email: "alex.johnson@acme.com"
    );

    private readonly HttpClient _httpClient;
    private readonly ILogger<Pipeline> _logger;

    public Pipeline(HttpClient httpClient, ILogger<Pipeline> logger)
    {
        _httpClient = httpClient;
        _logger = logger;
    }

    public async Task RunAsync(CancellationToken ct)
    {
        while (!ct.IsCancellationRequested)
        {
            await RunOnceAsync(ct);
            await Task.Delay(TimeSpan.FromSeconds(Config.PipelineIntervalSeconds), ct)
                .ConfigureAwait(false);
        }
    }

    private async Task RunOnceAsync(CancellationToken ct)
    {
        var contact = SampleContact;

        // Step 1: Validate
        try
        {
            var resp = await _httpClient.PostAsJsonAsync(Config.FluxUrl, contact, ct);
            resp.EnsureSuccessStatusCode();
            contact = await resp.Content.ReadFromJsonAsync<Contact>(ct) ?? contact;
        }
        catch (Exception ex) { _logger.LogWarning(ex, "Validate failed"); }

        // Step 2: Enrich
        try
        {
            var resp = await _httpClient.PostAsJsonAsync(Config.RiftUrl, contact, ct);
            resp.EnsureSuccessStatusCode();
            contact = await resp.Content.ReadFromJsonAsync<Contact>(ct) ?? contact;
        }
        catch (Exception ex) { _logger.LogWarning(ex, "Enrich failed"); }

        // Step 3: Score
        try
        {
            var resp = await _httpClient.PostAsJsonAsync(Config.SwellUrl, contact, ct);
            resp.EnsureSuccessStatusCode();
        }
        catch (Exception ex) { _logger.LogWarning(ex, "Score failed"); }
    }
}
